### ISSUE TITLE
Administrator not able to edit Owner of Channel

### ISSUE THREAD

**Issue by port22** [RocketChat/Rocket.Chat#17172]
Not explicitely a bug, but wouldn't it make sense to let the administrator users change the owners of private channels? Every other field is editable, but I can't find a reason why changing the owner should be prohibited for administrators; like when the only owner left.

It's possible to change the room to public, so ...

**Comment by ankar84** [comment]
> It's possible to change the room to public, so ...

Yes. That's only one workaround that I use for that cases.

- Convert Private Chat to public
- Find that Public Chat
- Add\Change owner
- Convert Public Chat to Private chat again in chat settings.

**Comment by port22** [comment]
> Add\Change owner

this is not possible either, although the room is public now.
There is no edit-icon next to the Owner field

Anyway, I actually wanted to open a discussion on implementing the possibility to edit the owner, instead of discussing workarounds

**Comment by ankar84** [comment]
> this is not possible either, although the room is public now.
> There is no edit-icon next to the Owner field

You can do it in regular (not Admin UI) channel settings. Set owner from Member list. Convert ro Private in Channel info - Edit

> Anyway, I actually wanted to open a discussion on implementing the possibility to edit the owner, instead of discussing workarounds

Here I totally agree with you. Such ability would be great to have.

**Comment by pierre-lehnen-rc** [comment]
You actually can change the owner, it's just not very straightforward:
 - Go to admin / Permissions
 - Edit the "Owner" role
 - Select the room you want to change the owner of by searching for it's name
 - Search for the user you want to add as owner and click on the 'Add' button.

**Comment by gabriellsh** [comment]
we should revise this so it's more straightforward in the future.

**Comment by port22** [comment]
> You actually can change the owner, it's just not very straightforward:

This issue is not for finding the way how to do it, but as a feature request to implement it the obvious way. thanks for explaining the workaround what ankar84 already did.

**Comment by pierre-lehnen-rc** [comment]
Of course, just pointing out that there's an existing option so the workaround is not needed.

**Comment by port22** [comment]
> https://github.com/RocketChat/Rocket.Chat/issues/17172#issuecomment-609738913

@pierre-lehnen-rc That already happened here; that's what I said.
you're bloating the thread.

**Comment by pierre-lehnen-rc** [comment]
Read my comment again. You do not need to convert the room to public.

**Comment by engelgabriel** [comment]
Let's make the option available. Since all events will be recorded on the event system, this should trigger a message on the channel so every member is aware of the change, who did it and when.

**Comment by vplb** [comment]
>>You actually can change the owner, it's just not very straightforward:
>>
>>Go to admin / Permissions
>>Edit the "Owner" role
>>Select the room you want to change the owner of by searching for it's name
>>Search for the user you want to add as owner and click on the 'Add' button.
We created a room, the ownership was handed from the initial owner to new owner, and the original owner still shows up in the room's information. If looking under the admin/permissions/owner role, it shows the new owner. 

This attempt to change ownership does not work for me. As admin, i try to add myself to a private group, and receive "invalid user - user is not a member of the room".

I think it's important that Admins have the ability to re-organize ownership. This shouldn't involve the original owner, as if they leave the room is unchangeable.

**Comment by audouts** [comment]
The method posted by @pierre-lehnen-rc worked for us. However, it's very unintuitive and there should be an easier way to make changes.

**Comment by tkyser-rodeo** [comment]
There definitely needs to be a better way to handle this. Admins should be able to have unfettered access to everything as they need it to reorganize and assist users. It's not like this is protecting users from having admins see what they do or whatever, you have either the audit panel and/or direct access to the DB anyway.

**Comment by emikolajczak** [comment]
Hi,
I have workaround to add room owner direct from DB.
`db.rocketchat_subscription.update({ "rid": "room-id", "u._id": "user-to-add-id" },  { $push: { "roles": "owner" } } );`

User who needs to be an owner must be a member of room before.

**Comment by julskinka** [comment]
None of the mentioned workarounds work for me. Old owner was still displayed under /admin/rooms/edit/<room_id>.
Had to make the change in mongo.

```
db.rocketchat_room.update(
  { _id: "<room_id>" },
  { $set:
    {
      "u" : {
        "_id" : "<user_id>",
        "username" : "<username>"
      }
    }
  }
)
```

**Comment by vplb** [comment]
It appears i have the same issue. A user created a channel and made it private, as admin i cannot see it in search. When i am in the admin GUI, in Rooms, i can pull it up but i cannot edit anything. 
Maybe i'm old school, but an admin proper should be able to take ownership of any room. 

As a sort of a work around, all of our automatically created rooms belong to our Admin account so we can manage them. The idea of having many unknown, un-editable rooms/channels is an organizational nightmare. Let alone if an owner leaves the system, then there's no way to re-take ownership without work arounds. 

None of the gui work arounds have worked, nor normal API calls. I'm going to attempt the DB queries. Thanks for those.

**Comment by onigoetz** [comment]
It seems that this workaround doesn't work anymore (3.17.0, 3.18.0):



> * Go to admin / Permissions
> * Edit the "Owner" role
> * Select the room you want to change the owner of by searching for it's name
> * Search for the user you want to add as owner and click on the 'Add' button.

What worked for me is:

> * Convert Private Chat to public
> * Find that Public Chat
> * Add\Change owner
> * Convert Public Chat to Private chat again in chat settings.

**Comment by ankar84** [comment]
> It seems that this workaround doesn't work anymore (3.17.0, 3.18.0):

Not confirm that. Just tested on 3.18.0
One thing that green balloon have variable instead of real data
![image](https://user-images.githubusercontent.com/4023037/131623832-907e8d46-4245-4b2f-a7fb-610ee8f9a73a.png)

**Comment by SamirSaidani** [comment]
> You actually can change the owner, it's just not very straightforward:
> 
>     * Go to admin / Permissions
> 
>     * Edit the "Owner" role
> 
>     * Select the room you want to change the owner of by searching for it's name
> 
>     * Search for the user you want to add as owner and click on the 'Add' button.

Doesn't work for Discussion, they don't appear as a room

**Comment by svenseeberg** [comment]
> Using version 4.6.1 I can only see the rooms I'm in.

That seems to be true as of 5.3.1. I tried to change the owner of a room I'm not a member of. The suggestions that come up when entering a room name consist only of rooms that I'm already a member of.

Also it seems the Mongo queries are not working anymore. If I find a solution, I'll let you know.

**Comment by himani-mehra** [comment]
Hi @sgohl  , I would love to take on this issue and contribute to the project. Could you please assign it to me if it's available? Looking forward to your response. Thanks!
